### PR TITLE
Make the mining process cancellable

### PR DESCRIPTION
**Pull request by onewhl**
Currently, there is a problem with the cancellation of the mining the refactorings in the specific commit. The research shows that the problem consists in canceling the background task, we don't actually get into the `onCancel()` method while mining process doesn't finish. It could be resolved by running `checkCancelled()` every 10 milliseconds.

Also, I renamed RefactoringsBundle to RefactorInsightBundle.

### CODE REVIEW

File: src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java [hunk 1]
```diff
@@ -187,28 +197,75 @@ public void mineAndWait(GitRepository repository) {
...
           public void run(@NotNull ProgressIndicator progressIndicator) {
-              CommitMiner.mineAtCommitTimeout(commit, innerState.refactoringsMap.map, project);
+            try {
+              runWithCheckCanceled(() -> {
+                    CommitMiner.mineAtCommitTimeout(commit, innerState.refactoringsMap.map, project);
+                    return null;
+                  },
+                  progressIndicator);
+            } catch (Exception e) {
+              e.printStackTrace();
```
Comment (TanVD) [inline src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java:219]: Probably you should use `Logger.error("Message", e)`, not just printStackTrace.

File: src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java [hunk 2]
```diff
@@ -187,28 +197,75 @@ public void mineAndWait(GitRepository repository) {
...
+            }
           }
         });
   }
 
+  /**
+   * Allows to interrupt a process which does not performs checkCancelled() calls by itself.
+   */
+  private <T> void runWithCheckCanceled(@NotNull final Callable<T> callable,
```
Comment (TanVD) [inline src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java:228]: You probably need here `Runnable`, not `Callable<T>`. Also I would move it to `IDEUtils` static class :)

File: src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java [hunk 3]
```diff
@@ -187,28 +197,75 @@ public void mineAndWait(GitRepository repository) {
...
+                                        @NotNull final ProgressIndicator indicator) throws ExecutionException {
+    while (true) {
+      indicator.checkCanceled();
+      try {
+        future.get(10, TimeUnit.MILLISECONDS);
+        return;
+      } catch (InterruptedException e) {
+        throw new ProcessCanceledException(e);
+      } catch (TimeoutException ignored) {
+        ignored.printStackTrace();
```
Comment (TanVD) [inline src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java:265]: Think about using `Logger` :)

File: src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java (hunk 1 above)
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java:219]: Yeah, we definitely should use Logger in the project! I will take care of it.

File: src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java (hunk 2 above)
Comment (onewhl) [reply src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java:228]: Exactly, thanks!

File: src/main/java/org/jetbrains/research/refactorinsight/processors/SingleCommitRefactoringTask.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  @Override
+  public void onCancel() {
+    super.onCancel();
+  }
+
+  @Override
+  public void onFinished() {
+    super.onFinished();
+    if (service.containsCommit(commit.getId().asString())) {
+      System.out.println(RefactorInsightBundle.message("finished"));
```
Comment (onewhl) [inline src/main/java/org/jetbrains/research/refactorinsight/processors/SingleCommitRefactoringTask.java:63]: logger.debug("...")

**Pull request merged**